### PR TITLE
Add jenkins_job_facts module

### PR DESCRIPTION
**Pull request by stpierre**
##### SUMMARY

Adds a new module, `jenkins_job_facts`, that can be used to discover existence and some basic information about existing Jenkins jobs.

##### ISSUE TYPE
 - New Module Pull Request

##### COMPONENT NAME

`jenkins_job_facts`

##### ANSIBLE VERSION

```
ansible 2.5.0 (jenkins-job-facts 15074622c1) last updated 2018/01/25 06:55:33 (GMT -500)
  config file = None
  configured module search path = [u'/home/stpierre/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /home/stpierre/devel/ansible/lib/ansible
  executable location = /home/stpierre/devel/ansible/bin/ansible
  python version = 2.7.14 (default, Jan 17 2018, 14:28:32) [GCC 7.2.1 20170915 (Red Hat 7.2.1-2)]
```

### CODE REVIEW

File: lib/ansible/modules/web_infrastructure/jenkins_job_facts.py [hunk 1]
```diff
@@ -0,0 +1,190 @@
...
+    password: hunter2
+  register: my_jenkins_job_facts
+'''
+
+RETURN = '''
+---
+jobs:
+  description: All jobs found matching the specified criteria
+  returned: success
+  type: string
```
Comment (resmo) [inline lib/ansible/modules/web_infrastructure/jenkins_job_facts.py]: is it a string or a list?

File: lib/ansible/modules/web_infrastructure/jenkins_job_facts.py [hunk 2]
```diff
@@ -0,0 +1,190 @@
...
+# Get facts about a single job in a folder using basic auth
+- jenkins_job_facts:
+    name: some-folder-name/some-job-name
+    user: admin
+    password: hunter2
+  register: my_jenkins_job_facts
+
+# Get facts about jobs matching a shell glob using basic auth
+- jenkins_job_facts:
+    name: some-job-*
```
Comment (resmo) [inline lib/ansible/modules/web_infrastructure/jenkins_job_facts.py]: shouldn't this use the glob param?

**Review by resmo — CHANGES_REQUESTED** [review]
needs_revision

File: lib/ansible/modules/web_infrastructure/jenkins_job_facts.py (hunk 2 above)
Comment (stpierre) [reply lib/ansible/modules/web_infrastructure/jenkins_job_facts.py]: Done.

File: lib/ansible/modules/web_infrastructure/jenkins_job_facts.py (hunk 1 above)
Comment (stpierre) [reply lib/ansible/modules/web_infrastructure/jenkins_job_facts.py]: Done

**Comment by resmo** [comment]
a short tests without valid credetials shows a traceback which should be catched.

~~~
* including generated source, if any, saving to: /home/resmo/.ansible_module_generated
* ansiballz module detected; extracted module source to: /home/resmo/debug_dir
***********************************
RAW OUTPUT

Traceback (most recent call last):
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 190, in <module>
    main()
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 186, in main
    run_module()
  File "/home/resmo/debug_dir/ansible_module_jenkins_job_facts.py", line 168, in run_module
    all_jobs = jenkins_conn.get_all_jobs()
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 814, in get_all_jobs
    jobs = [(0, "", self.get_info(query=JOBS_QUERY)['jobs'])]
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 551, in get_info
    Request(self._build_url(url))
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 430, in jenkins_open
    self.maybe_add_crumb(req)
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 303, in maybe_add_crumb
    self._build_url(CRUMB_URL)), add_crumb=False)
  File "/usr/local/lib/python2.7/dist-packages/jenkins/__init__.py", line 461, in jenkins_open
    raise JenkinsException('Error in request: %s' % (e.reason))
jenkins.JenkinsException: Error in request: [Errno 111] Connection refused
~~~

**Comment by resmo** [comment]
works for me otherwise, probably one nice feature would be to query by "color"

**Review by resmo — APPROVED** [review]

File: lib/ansible/modules/web_infrastructure/jenkins_job_facts.py [hunk 3]
```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
+#
+# Copyright: (c) Ansible Project
```
Comment (resmo) [inline lib/ansible/modules/web_infrastructure/jenkins_job_facts.py]: put your name/company in here

**Comment by resmo** [comment]
tested, works for me.

shipit

**Pull request merged**